iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -26,6 +26,44 @@ namespace iox
 {
 namespace cxx
 {
+namespace internal
+{
+/// @brief Type trait which verifies whether the passed type T has INVALID_STATE
+///        Overload chosen when INVALID_STATE is not present
+template <typename T, typename = void>
+struct has_invalid_state_member : std::false_type
```

Comment:
I would use in every code which is not strictly stl related our code style. Therefore could you please rename this struct in UpperCamelCase HasInvalidStateMember.
Additionally, could you please create here only a forward declaration of that type like:
```cpp
template<typename, typename>
struct HasInvalidStateMember;
```
and define the types in the inl file. From the point of template programming this would be the implementation which you defined here.

Reply:
Done